RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails.swift

```diff
@@ -0,0 +1,190 @@
+					.url
+					.appending(path: path)
+				await openURL(url)
+			}
+		case .verifyOnLedgerButtonTapped:
+			if case let .account(address, _) = state.address {
+				ledgerHardwareWalletClient.verifyAddress(of: address)
+			}
+			return .none
+		case .shareDismissed:
+			state.showShare = false
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .loadedTitle(.success(title)):
+			state.title = .success(title)
+			return .none
+		case let .loadedTitle(.failure(error)):
+			state.title = .failure(error)
+			return .none
+		case let .loadedQrImage(.success(image)):
+			state.qrImage = .success(image)
+			return .none
+		case let .loadedQrImage(.failure(error)):
+			state.qrImage = .failure(error)
+			return .none
+		}
+	}
+
+	private func loadTitle(state: inout State) -> Effect<Action> {
+		state.title = .loading
+		return .run { [address = state.address] send in
+			let result: TaskResult<String?> = switch address {
+			case let .account(address, _):
+				await TaskResult {
+					let account = try await accountsClient.getAccountByAddress(address)
+					return account.displayName.rawValue
+				}
+			case let .resource(address):
+				await TaskResult {
+					let resource = try await onLedgerEntitiesClient.getResource(address)
+					return resource.resourceTitle
+				}
+			case let .validator(address):
+				await TaskResult {
+					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
+					return entity.metadata?.name
+				}
+			case let .package(address):
+				await TaskResult {
+					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
+					return entity.metadata?.name
+				}
+			case let .resourcePool(address):
+				await TaskResult {
+					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
+					return entity.metadata?.name
+				}
+			case let .component(address):
+				await TaskResult {
+					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
+					return entity.metadata?.name
+				}
+			case let .nonFungibleGlobalID(globalId):
+				await TaskResult {
+					let resource = try await onLedgerEntitiesClient.getResource(globalId.resourceAddress)
+					return resource.resourceTitle
+				}
+			}
+			await send(.internal(.loadedTitle(result)))
+		}
+	}
+
+	private func loadQrCode(state: inout State) -> Effect<Action> {
+		state.qrImage = .loading
+		let content = QR.addressPrefix + state.address.address
+		return .run { send in
+			let result = await TaskResult {
+				try await qrGeneratorClient.generate(.init(content: content))
+			}
+			await send(.internal(.loadedQrImage(result)))
+		}
+	}
+}
+
+// MARK: - Helpers
+
+private extension OnLedgerEntity.Resource {
+	var resourceTitle: String? {
```

Comment:
We used to have something with a very similar name, which returned `metadata.symbol ?? metadata.name`, but I can't find it right now. If it still exists somewhere it might be a bit confusing.

Reply:
Probably [this](https://github.com/radixdlt/babylon-wallet-ios/blob/main/RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient%2BCreateEntity.swift#L582-L585) `title` defined on `OnLedgerEntity.Metadata`. 

The one I defined is private anyway, and is an improved version of it. `smartResourceTitle`?

Reply:
I wouldn't say it's improved, it's different, and only makes sense here

Reply:
So I would just make it a private computed property in the View

Reply:
It is already a private computed property, but it is an extension of the `OnLedgerEntity.Resource`, and not of the View, since it is used in multiple places
```Swift
try await onLedgerEntitiesClient.getResource(address).resourceTitle
try await onLedgerEntitiesClient.getResource(globalId.resourceAddress).resourceTitle
```

Given it is already private, I will leave it as is.